main.py

```diff
@@ -0,0 +1,87 @@
+import operator
+from typing import Callable
+Self = 'Matrix'
+
+
+# Purpose: Create a class to represent a two-dimensional array
+class Matrix:
+    """A class to represent a two-dimensional array"""
+    def __init__(self, matrix: tuple[tuple[float, ...], ...]):
+        self._matrix = matrix
+
+    def __str__(self):
+        """Return a string representation: ((1, 2), (3, 4))"""
+        return str(self._matrix)
+
+    def __repr__(self):
+        """ return: a string representation: Matrix(((1, 2), (3, 4)))"""
+        return f'Matrix({self._matrix})'
+
+    @property
+    def tuples(self):
+        """Return the matrix"""
+        return self._matrix
+
+    def _addition_subtraction(self, other: Self, symbol: Callable[[float, float], float]) -> Self:

```

Comment:
change `symbol` to `action`

Reply:
the name of the function can be more generic, because it can handle all action between 2 numbers